examples/pytorch/ogc/ogc.py

```diff
@@ -0,0 +1,116 @@
+import argparse
+import time
+
+import scipy.sparse as sp
+
+import torch
+import torch.nn.functional as F
+
+from dgl import AddSelfLoop
+from dgl.data import CiteseerGraphDataset, CoraGraphDataset, PubmedGraphDataset
+from utils import (
+    LinearNeuralNetwork,
+    sparse_mx_to_torch_sparse_tensor,
+    symmetric_normalize_adjacency,
+)
+
+
+# Training settings.
+decline = 0.9  # the decline rate
+lr_sup = 0.001  # the learning rate for supervised loss
+lr_clf = 0.5  # the learning rate for the used linear classifier
+beta = 0.1  # the moving probability that a node moves to its neighbors
+max_sim_rate = 0.995  # the max label prediction similarity between iterations
+max_patience = 2  # the tolerance for consecutively similar test predictions
+
+
+def update_embeds(embeds, graph, label_idx_mat):
+    global lr_sup
+    # Update classifier's weight by training a linear supervised model.
+    pred_labels, clf_weight = linear_clf.update_weight(embeds, graph, lr_clf)
+    labels = F.one_hot(graph.ndata["label"]).float()
+
+    # Update the smoothness loss via LGC.
+    embeds = torch.spmm(lazy_adj.to(device), embeds)
+
+    # Update the supervised loss via SEB.
+    deriv_sup = 2 * torch.mm(
+        torch.sparse.mm(label_idx_mat, -labels + pred_labels), clf_weight
+    )
+    embeds = embeds - lr_sup * deriv_sup
+
+    lr_sup = lr_sup * decline
+    return embeds
+
+
+def OGC(linear_clf, embeds, graph, label_idx_mat):
```

Comment:
This is the training function but the function name is `OGC` which looks like a model class in terms of PyTorch convention.  This is confusing.  Consider renaming it to `train`.

Reply:
Thanks. As suggested, we have renamed it to 'train(..)', in the main file (i.e., the '[train.py](https://train.py/)' file).